Login Start Pages dropdown list - Clouds menus

GregP opened the pull request:
**NOTE: This PR requires MiqShortcut.seed to be run to re-populate database table.**

**NOTE: In order to check Datawarehouse and Monitor Alerts shortcuts and menu display settings need to change in Configuration => Server => Advanced tab, see screen shots below for actual settings change/addition if needed.**

https://bugzilla.redhat.com/show_bug.cgi?id=1331327

At Login Start Page drop down list in My Settings => Visual tab is not in sync with recent UI menu navigation changes/updates.

Code updates provided via multiple commits, each matching UI menu section.

**Screen shots of Start Page drop down list PRIOR to code fixes:**
![all start pages list prior to update 1](https://cloud.githubusercontent.com/assets/552686/25762618/4e6528c0-3194-11e7-8019-862186f882f2.png)


![all start pages list prior to update 2](https://cloud.githubusercontent.com/assets/552686/25762567/176a4030-3194-11e7-8e49-5fbf5ffff4a4.png)

![all start pages list prior to update 3](https://cloud.githubusercontent.com/assets/552686/25762574/1c432e78-3194-11e7-85d5-fbedd9256c18.png)

![all start pages list prior to update 4](https://cloud.githubusercontent.com/assets/552686/25762581/210ad0be-3194-11e7-81e7-a005e7732aa3.png)

![all start pages list prior to update 5](https://cloud.githubusercontent.com/assets/552686/25762583/24b914c8-3194-11e7-9c63-c50b457e71d2.png)



**Screen shots of Start Page drop down AFTER code fixes:**

![start page at login cloud intel short cuts post fix](https://cloud.githubusercontent.com/assets/552686/26176925/9f55e9dc-3b0c-11e7-8ef2-c8ad5d8b773e.png)

![start page at login services short cuts post fix](https://cloud.githubusercontent.com/assets/552686/26176932/a63e6116-3b0c-11e7-9ab6-bf11a4af9f63.png)

![start page at login compute clouds short cuts post fix](https://cloud.githubusercontent.com/assets/552686/26176946/b0ff0128-3b0c-11e7-94ac-8bc731d92f9d.png)

![start page at login compute infrastructure short cuts post fix](https://cloud.githubusercontent.com/assets/552686/26176952/b694da54-3b0c-11e7-9c3f-5d9847b9bbdd.png)

![start page at login compute containers short cuts post fix](https://cloud.githubusercontent.com/assets/552686/26176955/bb9cb3be-3b0c-11e7-8a3e-dd0986058f3b.png)

![start page at login configuration short cuts post fix](https://cloud.githubusercontent.com/assets/552686/26176963/c5ff998e-3b0c-11e7-9e9a-6df583c879ab.png)

![start page at login networks short cuts post fix](https://cloud.githubusercontent.com/assets/552686/26176969/cc153d60-3b0c-11e7-8733-885289c59390.png)

![start page at login middleware short cuts post fix](https://cloud.githubusercontent.com/assets/552686/26176974/d0dd0c6a-3b0c-11e7-8a49-ae92bfadd91a.png)

![start page at login storage short cuts post fix](https://cloud.githubusercontent.com/assets/552686/26176983/d9b036a0-3b0c-11e7-811e-58fbafc65f97.png)

![start page at login control short cuts post fix](https://cloud.githubusercontent.com/assets/552686/26176992/e275e690-3b0c-11e7-9d98-a368dbd9de77.png)

![start page at login automation short cuts post fix](https://cloud.githubusercontent.com/assets/552686/26177001/eadc2d76-3b0c-11e7-92ad-c48964ef7565.png)

![start page at login optimize short cuts post fix](https://cloud.githubusercontent.com/assets/552686/26177009/f2025562-3b0c-11e7-99e2-c9c448e76f08.png)


**NOTE: In order to see Datawarehouse and Monitor Alerts menus display in UI, change/add following settings in (top right corner) Configuration => Server => Advanced Tab:**

![datawarehouse and monitor alerts settings change in config server advanced tab](https://cloud.githubusercontent.com/assets/552686/25873765/92c89eba-34c4-11e7-9d9d-6435d597052b.png)

UI Datawarehouse menu displayed post settings set/save:

![datawarehouse and monitor alerts menus post settings change](https://cloud.githubusercontent.com/assets/552686/25873776/a19a8250-34c4-11e7-9259-234eedd12de4.png)

GregP commented:
@h-kataria Please review.

dclarizio commented:
@chessbyte please assign to @h-kataria, she can review this.  Thx, Dan

GregP commented:
@h-kataria Added Datawarehouse and Monitor Alerts, per discussion. Added notes and screen shots to assist in settings addition/update to display these menus in UI. Ready for review.

h-kataria commented on db/fixtures/miq_shortcuts.yml:
```diff
@@ -4,257 +4,453 @@
-  :startup: true
-- :name: vms_mytemplates
-  :description: Services / Workloads / Templates & Images
-  :url: /vm_or_template/explorer?accordion=templates_images_filter
```
@GregP we need to keep all VM explorer related start page entries that allows user to be able to set a specific accordion as their start page. See my screenshot below where i have set "Services / Workloads/ Templates & Images" accordion to be my start page.
![screenshot from 2017-05-13 13-27-00](https://cloud.githubusercontent.com/assets/3450808/26027686/421b7b62-37e0-11e7-8e57-a4e6f4e380a8.png)

GregP commented:
@h-kataria We do not have those in the defaults file? One of my commits deleted Workloads/xxx shortcuts, I can put those back, just don't see them listed in the defaults (matching) file?

GregP commented:
@h-kataria Services short cuts re-instated, ready for review.

GregP commented:
@h-kataria All short cuts should be up to date now, ready for review.

GregP commented:
@h-kataria Shortcuts amended to streamline Services, Compute Clouds Instances and Compute Clouds Infrastructure, ready for review.

h-kataria commented on db/fixtures/miq_shortcuts.yml:
```diff
@@ -53,208 +53,439 @@
   :url: /vm_or_template/explorer?button=vm_miq_request_new
   :rbac_feature_name: vm_miq_request_new
-  :startup: false
+  :startup: true
```
@GregP any idea why this is changed to true, i don't remember what is the use of setting startup attribute here, is that being used to ignore the shortcut entry during seeding or something. Can you please check.

GregP commented:
@h-kataria I think I switched it to true while applying all the later changes/additions and testing yesterday. Changed it back to false, its original setting. Ready for review.

h-kataria commented:
@chessbyte looks good to be merged

simaishi commented:
@GregP There are a few conflicts backporting this to Fine branch. Can you create a PR for FIne branch?

simaishi commented:
Backported to Fine via https://github.com/ManageIQ/manageiq/pull/15734
